pages/proposals.jsx

```diff
@@ -2,21 +2,553 @@ var React = require('react');
+      );
+    }
+    return (
+      <input maxLength={this.props.maxlength} name={this.props.name || ""} onClick={this.props.onClick || function() {}} onChange={this.props.updateFunction} id={this.props.for} type={this.props.type}/>
+    );
+  }
+});
+
+var InputCombo = React.createClass({
+  updateFunction: function(e) {
+    var errorElement = document.querySelector("#" + this.props.name + "Error");
+    if (errorElement) {
+      errorElement.classList.remove("show");
+    }
+    if (this.props.wordcount || this.props.wordcount === 0) {
+      var value = e.target.value.trim();
+      var wordcount = this.props.wordcount;
+      if (value) {
+        wordcount = this.props.wordcount - value.split(/\s+/).length;
+      }
+      this.setState({
+        wordcount: wordcount
+      });
+    }
+  },
+  getInitialState: function() {
+    if (this.props.wordcount || this.props.wordcount === 0) {
+      return {
+        wordcount: this.props.wordcount
+      };
+    } else {
+      return {};
+    }
+  },
+  render: function() {
+    var hasWordcount = (this.props.wordcount || this.props.wordcount === 0);
+    var inputClassName = "";
+    var wordcountClassName = "word-count";
+    var inputContainerClassName = "input-container";
+    if (hasWordcount) {
+      inputClassName = "has-wordcount";
+      inputContainerClassName += " input-container-has-wordcount";
+      if (this.state.wordcount < 0) {
+        wordcountClassName += " negative";
+      }
+    }
+    var wordcountElement;
+    if (hasWordcount) {
+      wordcountElement = (<p className={wordcountClassName}>{this.state.wordcount}</p>);
+    }
+    var labelElement = (
+      <label id={this.props.for + "Link"} className={this.props.className} htmlFor={this.props.for}>
+        {this.props.children}
+      </label>
+    );
+    var errorElement = (<div></div>);
+    if (this.props.errorMessage) {
+      errorElement = (
+        <div id={this.props.name + "Error"} className="error-message">{this.props.errorMessage}</div>
+      );
+    }
+    var topLabel;
+    var bottomLabel;
+    if (this.props.type !== "checkbox" && this.props.type !== "radio") {
+      topLabel = labelElement;
+    } else {
+      bottomLabel = labelElement;
+    }
+    return (
+      <div className={this.props.for + "-container input-combo " + this.props.type + "-container"}>
+        {topLabel}
+        <div className={inputContainerClassName}>
+          {wordcountElement}
+          <RealInput onClick={this.props.onClick || function() {}} className={inputClassName} maxlength={this.props.maxlength || 1650} updateFunction={this.updateFunction} for={this.props.for} type={this.props.type} name={this.props.name}/>
+        </div>
+        {bottomLabel}
+        {errorElement}
+      </div>
+    );
+  }
+});
 
 var Proposals = React.createClass({
+  submit: function(fields) {},
```

Comment:
I need t flesh out this function to grab the data from the fields based on field names passed in. Then send that to the /add-session endpoint and ensure it enters the google sheet I'm setting up now.